File: Android/Java/Asg2 - Ice Cream App/app/src/main/java/com/example/asg2icecreamapp/OrderHistoryActivity.java
package com.example.asg2icecreamapp;


import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

public class OrderHistoryActivity extends AppCompatActivity {
    ListView listView;
    ArrayList<OrderItem> orders;
    ArrayList<String> stringOrders;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_order_history);

        setTitle("Order History");
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        //setup the ListView
        Intent i = getIntent();
        orders = (ArrayList<OrderItem>) i.getSerializableExtra("OrdersKey");
        //Log.d("DEBUG", orders.toString());
        stringOrders = new ArrayList<String>();
        for (OrderItem item: orders)
            stringOrders.add(item.toString());
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(
                this,
                android.R.layout.simple_list_item_1,
                android.R.id.text1,
                stringOrders
        );
        listView = findViewById(R.id.listView);
        listView.setAdapter(adapter);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                this.finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: Android/Kotlin/Asg3 Ice Cream App/app/src/main/java/com/example/asg3icecreamapp/OrderHistoryActivity.kt
package com.example.asg3icecreamapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_order_history.*

class OrderHistoryActivity : AppCompatActivity() {
    private var ordersSTR = ArrayList<String>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_order_history)

        title = "Order History"
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val i = intent
        val orders = i.getSerializableExtra("OrdersKey") as ArrayList<OrderItem>
        for(order in orders) {
            ordersSTR.add(order.toString())
        }

        val adapter = ArrayAdapter<String> (
            this,
            android.R.layout.simple_list_item_1,
            android.R.id.text1,
            ordersSTR)

        orderHistoryListView.adapter = adapter
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                this.finish()
                return true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
